client/components/ApplyRandomPrices.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import { gql, useMutation } from "@apollo/client";
+import { Layout, Button, Banner, Toast, Stack, Frame } from "@shopify/polaris";
+import { Loading } from "@shopify/app-bridge-react";
+
+// GraphQL mutation that updates the prices of products
+const UPDATE_PRICE = gql`
+  mutation productVariantUpdate($input: ProductVariantInput!) {
+    productVariantUpdate(input: $input) {
+      product {
+        title
+      }
+      productVariant {
+        id
+        price
+      }
+    }
+  }
+`;
+
+function ApplyRandomPrices(props) {
+  const { selectedItems, onUpdate } = props;
+  const [hasResults, setHasResults] = useState(false);
+
+  const showToast = hasResults && (
```

Comment:
Nit: should we name this just `toast`? `showToast` makes my brain think it's a boolean.